app/jobs/journaled/delivery_job.rb

```diff
@@ -12,26 +12,11 @@ class DeliveryJob < ApplicationJob
       raise KinesisTemporaryFailure
     end
 
-    UNSPECIFIED = Object.new
-    private_constant :UNSPECIFIED
-
-    def perform(serialized_event:, partition_key:, stream_name: UNSPECIFIED, app_name: UNSPECIFIED)
-      @serialized_event = serialized_event
-      @partition_key = partition_key
-      if app_name != UNSPECIFIED
-        @stream_name = self.class.legacy_computed_stream_name(app_name: app_name)
-      elsif stream_name != UNSPECIFIED && !stream_name.nil?
-        @stream_name = stream_name
-      else
-        raise(ArgumentError, 'missing keyword: stream_name')
-      end
-
-      journal!
-    end
+    def perform(*events, **legacy_kwargs)
+      events << legacy_kwargs if legacy_kwargs.present?
+      @kinesis_records = events.map { |e| KinesisRecord.new(**e.delete_if { |_k, v| v.nil? }) }
```

Comment:
I'm wrapping these with `KinesisRecord` structs to get the same kwarg argument safety as before, but I could be convinced to just hold onto them as hash records. All the job effectively does is rename `:serialized_event` to `:data` before handing off to kinesis, and the kinesis SDK has its own argument safety checks.